osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -304,20 +293,22 @@ private void load(LocalisationManager localisation)
                 titleBinding.BindValueChanged(_ => setMetadata(metadata.Source));
                 artistBinding.BindValueChanged(_ => setMetadata(metadata.Source), true);
 
+                starDifficulty = difficultyCache.GetBindableDifficulty(beatmapInfo, (cancellationSource = new CancellationTokenSource()).Token);
+                starDifficulty.BindValueChanged(s =>
+                {
+                    difficultyColourBar.Current.Value = s.NewValue ?? default;
```

Comment:
Why are these done here rather than direct binds?

Reply:
`StarRatingDisplay` and `DifficultyColourBar` both accept type `StarDifficulty`, while the returned bindable is of type `StarDifficulty?`.

Reply:
Yet they handle weird defaults locally? Wouldn't it make more sense to either make `default` used everywhere or have them support nullability? Not sure.

Reply:
Not sure either, `StarRatingDisplay` supporting nullability sounds wrong unless you're fine with making it display 0.00 or hide itself, I would expect a null value on a component to display something indicating its value.

Reply:
Well as it stands, you're just using `default`. Doesn't that give the same behaviour?

Reply:
The `default` there won't actually be hit as the value change event doesn't invoke once immediately, I should've added a `Debug.Assert` instead.

Reply:
So the display elements have a hidden behaviour for when they don't have an initial value... which can only be reached once ever? That seems really weird.

Reply:
Only for the difficulty colour bar which is currently visible, can hide that until a star difficulty is calculated then.

Reply:
While I was going through making all components accept `StarDifficulty?` instead, it turns out the following will also be required to bind, unfortunately:

![CleanShot 2021-08-17 at 18 47 31](https://user-images.githubusercontent.com/22781491/129758441-59038aa2-9094-4862-a05e-c44d3e2b8710.png)

I'll change to an assert and leave the rest for another PR, as otherwise things will get out of scope given that `StarRatingDisplay` only accepts `StarDifficulty` itself.